pkg/operator/csi/csistorageclasscontroller/csi_storageclass_controller.go

```diff
@@ -61,8 +64,15 @@ func NewCSIStorageClassController(
 	kubeClient kubernetes.Interface,
 	informerFactory informers.SharedInformerFactory,
 	operatorClient v1helpers.OperatorClient,
+	operatorInformer opinformers.SharedInformerFactory,
 	eventRecorder events.Recorder,
 	optionalStorageClassHooks ...StorageClassHookFunc) factory.Controller {
+	clusterCSIDriverLister := operatorInformer.Operator().V1().ClusterCSIDrivers().Lister()
+	evaluator := NewStorageClassStateEvaluator(
+		kubeClient,
+		clusterCSIDriverLister,
+		eventRecorder,
+	)
```

Comment:
You will have to add this informer in `WithInformers`  call of this controller, so as we wait for it to be synced before performing any operation.

Reply:
Added, thanks, this might explain some issues I ran into during testing.